Feature/f06

Krylovchan opened the pull request:
выполнена задача Implement News importer for VK #6
Сделано получение постов со стены паблика в VK
Сделано получение информации о пабллике VK
Реализованы юнит-тесты для VkClient

Krylovchan committed:
Реализация чтения информации о группе, добавлены авто тесты

Krylovchan committed:
commit 2

сделал не полностью метод для постов
проверил в test, что принимает правильные значения
не понял как camel до конца переделать

Krylovchan committed:
final commit

все поля, написал тест, исправил camel
возвращает все необходимые поля
в attachments реализовал ссылку, фото, альбом, видео, документ, все считывает нормально
перевел вроде все в camel case
написал тест проверяющий нормальное возвращение данных

andreygolubkow commented:
@Krylovchan  В проект с тестами нужно добавить Readme.md и там написать, как добавить API ключ. Так как эти тесты работать без него не будут

andreygolubkow commented on TaktTusur.Media.Clients.VkApi.Tests/VkApiClientTests.cs:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    [Test]
+    public async Task  Test1()
```
Нужно нормальное название для метода вместо Test1. Это же для следующих методов

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/GroupByIdResponse.cs:
```diff
@@ -0,0 +1,47 @@
+
+namespace TaktTusur.Media.Clients.VkApi;
+
+public class GroupByIdResponse
```
1 файл - 1 класс.
Если два класса в одном файле, нужно один из них перенести в другой файл

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/IVkApiClient.cs:
```diff
@@ -2,5 +2,6 @@ namespace TaktTusur.Media.Clients.VkApi;
-    
-}
\ No newline at end of file
+    Task<VkGroupInfo> GetGroupInfoAsync(CancellationToken cancellationToken);
```
Нужны комменты к методам, что они делают

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+            info.GroupPhoto200 = result.Response.Groups[0].Photo200;
+            info.GroupIsClosed = result.Response.Groups[0].IsClosed;
+        }
+        else Console.WriteLine("ошибка в блоке информации о группе");
```
Нельзя в консоль писать ошибку, так как консоли может и не быть. Нужно выбросить исключение

andreygolubkow replied:
https://upread.ru/art.php?id=656

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/GroupByIdRequest.cs:
```diff
@@ -0,0 +1,17 @@
+﻿using RestEase;
+using System.Text.Json.Serialization;
```
Эти два импорта(сверху) не используются. Их нужно убрать.

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/GroupByIdResponse.cs:
```diff
@@ -0,0 +1,47 @@
+public class GroupByIdResponse
+{
+    public Response Response { get; set; }
+    public Error GroupInfoError { get; set; }
```
Почему здесь используется класс `Error` из `Elasticsearch.Net` ? Скорее всего нужно было создать свой класс и туда писать ошибку.

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/GroupByIdResponse.cs:
```diff
@@ -0,0 +1,47 @@
+        public List<Group> Groups { get; set; }
+
+        [JsonPropertyName("profiles")]
+        public List<object> Profiles { get; set; }
```
Почему здесь `List<object>` ? Что должно лежать в этом свойстве ? Если оно не используется, то нужно его убрать

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+   
+    private readonly VkApiOptions _options;
+    private readonly RestClientOptions restClientOptions;
+    private readonly GroupByIdRequest GroupRequest;
```
Приватные поля пишутся с маленькой буквы, перед названием переменной должно стоять `_`

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+     public async Task<VkPost> GetPostsAsync(CancellationToken cancellationToken)
+     {
+        //чтобы вытащить id группы
+        var Group = new VkApiClient(_options, WallRequest.Domain);
```
Локальные переменные пишутся с маленькой буквы

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+
+    }
+        
+    public async Task<VkGroupInfo> GetGroupInfoAsync(CancellationToken cancellationToken)
```
Сигнатура метода не совпадает с тем, что указано в задании. 
Метод должен принимать id группы и cancellation token, а у тебя id группы передается в класс при создании. Получается так, что этот класс может быть использован только для одной группы - это не очень. 
![image](https://github.com/takttusur/takt-media/assets/15832039/a19302cd-8262-458f-868d-4e1c9bc9d1fe)

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+        return info;
+    }
+
+     public async Task<VkPost> GetPostsAsync(CancellationToken cancellationToken)
```
Сигнатура метода не совпадает с тем, что в задании. Метод должен принимать всю необходимую информацию для запроса в качестве аргумента.

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+            Version = "5.154",
+            AccessToken = _options.Key,
+            Domain = groupId,
+            Count = 5,
```
Количество запрашиваемых постов должно быть изменяемым, а не фиксированным, как у тебя.

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+        }
+        else 
+        {
+            TestResult.Count = 10;
```
Для чего ты выставляешь TestResult.Count = 10 ?

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/VkApiClient.cs:
```diff
@@ -1,5 +1,240 @@
+        else 
+        {
+            TestResult.Count = 10;
+            Console.WriteLine("ошибка в блоке посты");
```
Тут нельзя писать в консоль. Нужно выбрасывать исключение

Krylovchan committed:
добавил Readme.md,  изменил название тестов,. разбил Responses на файлы, исправил вывод ошибок

В Readme.md написал как получить Api ключ и как добавить его в код ;
написал названия для тестов, отражая их назначение ;
Response для постов, и для информации о группе, разбил на 1 файл = 1 класс ;
ошибки о доступах к группе или постам со стены паблика, теперь выводятся не в консоль, а как исключение;
написал комменты к методам, описывающие, что делает метод.

andreygolubkow commented on TaktTusur.Media.Clients.VkApi/GroupInfoResponse/GroupByIdResponse.cs:
```diff
@@ -0,0 +1,12 @@
+public class GroupByIdResponse
+{
+    public Response Response { get; set; }
+    public Error GroupInfoError { get; set; }
```
Почему используется error из библиотеки elasticsearch ?

Krylovchan committed:
Изменена сигнатура методов GetPostsAsync и GetGroupInfoAsync, написан класс VkApiException, написаны тесты для выброса ошибок

1. метод GetGroupInfoAsync теперь дополнительно запрашивает id группы, запрос также формируется теперь в методе
2. метод GetPostsAsync теперь дополнительно запрашивает id группы и количество выводимых постов
3. VkApiException наследуется от Exception и дополнительно реализует код ошибки
4. написаны 2 теста на проверку выбрасывания ошибок в методах GetGroupInfoAsync и GetPostsAsync

Krylovchan committed:
удалил using

andreygolubkow committed:
Merge branch 'develop' into feature/F06

andreygolubkow reviewed (commented):
MERGED
